dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/AbstractJdbcEventAnalyticsManager.java

```diff
@@ -232,7 +232,19 @@ else if ( !params.hasPeriods() && params.hasFilterPeriods() )
                 ProgramIndicator in = (ProgramIndicator) queryItem.getItem();
 
                 String asClause = " as " + quote( in.getUid() );
-                columns.add( "(" + programIndicatorService.getAnalyticsSql( in.getExpression(), in, params.getEarliestStartDate(), params.getLatestEndDate() ) + ")" + asClause );
+
+                String piSql = programIndicatorService.getAnalyticsSql( in.getExpression(), in, params.getEarliestStartDate(), params.getLatestEndDate() );
+
+                if ( queryItem.hasRelationshipType() )
+                {
+                    piSql += " AND ax.tei in (select tei.uid from " + "trackedentityinstance tei"
+                        + " LEFT JOIN relationshipitem ri on tei.trackedentityinstanceid = ri.trackedentityinstanceid "
+                        + " LEFT JOIN relationship r on r.relationshipid = ri.relationshipid"
```

Comment:
We also need to add to the select a check that the parent TEI is on either the r.from_relationshipitemid side or the r.to_relationshipitemid side of the relationship, and the subquerys TEI should be on the opposite side of the same relationship. 
In addition, we should probably add support for all the different types of relationships now - this would probably require a bit more work, potentially a new method to handle the relatinship joins. The relationships might connect events, enrollments or TEI - and the relationship type definiton will contain the information on what types are joined for this relationship type. Potentially we need to have different SQLs based on the relationship type definiton, and also the analytics type of the program indicator we want to select.